### PR TITLE
Enable strict mypy checks

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

Enable strict mypy checks
replaces #53

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionalit)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] Local tests pass.
- [x] There is no commented out code in this PR.
- [x] The code has been formatted using Black (`make black`)
- [x] Tests have been added to verify that the new code works.

### CODE REVIEW

**Comment by ludeeus** [comment]
@KapJI can you take a look at this?

File: awesomeversion/awesomeversion.py [hunk 1]
```diff
@@ -14,17 +15,18 @@
...
 
 class _AwesomeVersionBase(str):
     """Base class for AwesomeVersion to allow the usage of the default JSON encoder."""
 
-    def __init__(self, string):
+    def __init__(self, string: str):
         str.__init__(string)
 
-    def __new__(cls, version, _=None):
+    def __new__(cls, version: str, _: Optional[Any] = None) -> "_AwesomeVersionBase":
```
Comment (KapJI) [inline awesomeversion/awesomeversion.py:29]: You can add `from __future__ import annotation` to use unquoted type.
```suggestion
    def __new__(cls, version: str, _: Optional[Any] = None) -> _AwesomeVersionBase:
```

**Review by KapJI — APPROVED** [review]
LGTM, thanks!

In #53 I also added basic isort config, it might be useful to have consistent output for isort and black.

File: awesomeversion/awesomeversion.py (hunk 1 above)
Comment (ludeeus) [reply awesomeversion/awesomeversion.py:29]: `from __future__ import annotation` can only be used from python 3.8, this library has support for 3.6 and 3.7 as well.

**Comment by ludeeus** [comment]
> In #53 I also added basic isort config, it might be useful to have consistent output for isort and black.

 I did not see that, added that as well now :+1:

**Comment by ludeeus** [comment]
Thanks for looking it over @KapJI :+1:

**Pull request merged**